cpp/neuralnet/nneval.cpp

```diff
@@ -948,6 +959,48 @@ void NNEvaluator::evaluate(
 
 }
 
+void NNEvaluator::evaluate_symmetry(
+  Board& board,
+  const BoardHistory& history,
+  Player nextPlayer,
+  MiscNNInputParams& nnInputParams,
+  NNResultBuf& buf,
+  bool skipCache,
+  bool includeOwnerMap,
+  Rand& rand
+) {
+  assert(nnInputParams.numSymmetriesToSample > 1);
+  Hash128 nnHash = NNInputs::getHash(board, history, nextPlayer, nnInputParams);
+  if(skipCache || nnCacheTable == NULL || !nnCacheTable->get(nnHash,buf.result)) {
```

Comment:
I don't like the nnCache lookup in this if statement, it's too complex. and not clear why after getting the cache it still needs to call evaluate. Can it just be a block above it if(not skipcache and cachehit) return

Reply:
Because we're only calling evaluate on a cache miss. If it's in the cache, !get() will be false.

Reply:
I could add a comment above it if that helps?

Reply:
yes, the !get will be false, and it will go to else to evaluate, even though it's already in buf, right?

Reply:
Err right. What's going on there. I need to review it when I have a few minutes.

The logic maybe should be a 2nd check after this one that if it's in the buf already don't eval at all, just return? Or is there no harm in calling evaluate() if it's already in buf, since if it's cached there's no nn eval wasted? But yeah extra processing steps we can skip we should skip.

Is it possible for cache to hit but ownership map need updated? If I recall evaluate() does more than just fetch the cache.